openedx/core/djangoapps/credit/signals.py

```diff
@@ -70,3 +75,46 @@ def listen_for_grade_calculation(sender, username, grade_summary, course_key, de
                     api.set_credit_requirement_status(
                         username, course_id, 'grade', 'grade', status="failed", reason={}
                     )
+
+
+def tag_course_content_with_partition_scheme(course_key, partition_scheme):
+    """ Create user partitions with provided partition scheme and tag credit
+    blocks for the given course with those user partitions.
+
+    Args:
+        course_key (CourseKey): Identifier for the course.
+        partition_scheme (str): Name of the user partition scheme
+
+    """
+    partition = None
+    # create partition with provided partition scheme
+    if partition_scheme == 'verification':
+        partition = VerificationPartitionScheme()
+
+    if partition is None:
+        log.error(u'No user partition found with scheme "%s".', partition_scheme)
+        # TODO: maybe raise a 'NoSuchUserPartitionError' exception if not user partition is present with provide 'partition_scheme'.
```

Comment:
I think you can use `UserPartition.get_scheme(partition)` to dynamically load the partition scheme.  This will use Stevedore (a plugin manager) to discover and load the partition scheme by its name, raising a `NoSuchUserPartitionError` if one is not found (which you can re-raise in this function).

I like that this method is generic; I think that will make it easier to maintain and re-use.